src/pages/workspaces/workspace/Data.test.ts

```diff
@@ -0,0 +1,137 @@
+import { DeepPartial } from '@terra-ui-packages/core-utils';
+import { act, render, screen } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import { Ajax } from 'src/libs/ajax';
+import { LeoAppStatus, ListAppResponse } from 'src/libs/ajax/leonardo/models/app-models';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+import { testOnlyExports } from 'src/pages/workspaces/workspace/Data';
+import { asMockedFn } from 'src/testing/test-utils';
+import { defaultAzureWorkspace } from 'src/testing/workspace-fixtures';
+
+import { StorageDetails } from './useWorkspace';
+
+const { WorkspaceData } = testOnlyExports;
+
+type WorkspaceContainerExports = typeof import('src/pages/workspaces/workspace/WorkspaceContainer');
+jest.mock('src/pages/workspaces/workspace/WorkspaceContainer', (): WorkspaceContainerExports => {
+  return {
+    ...jest.requireActual<WorkspaceContainerExports>('src/pages/workspaces/workspace/WorkspaceContainer'),
+    wrapWorkspace: jest.fn().mockImplementation((_opts) => (wrappedComponent) => wrappedComponent),
+  };
+});
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): AjaxExports => {
+  return {
+    ...jest.requireActual<AjaxExports>('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+describe('WorkspaceData', () => {
+  // When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
+  // this type belongs there.
+  interface WorkspaceDataProps {
+    namespace: string;
+    name: string;
+    workspace: WorkspaceWrapper;
+    refreshWorkspace: () => void;
+    storageDetails: StorageDetails;
+  }
+
+  // SIFERS setup, see: https://medium.com/@kolodny/testing-with-sifers-c9d6bb5b362
+  function setup({
```

Comment:
It would be nice to declare types for this function's parameters and return value.

```ts
type SetupOptions = { ... };

type SetupResult = { ... };

function setup(opts: SetupOptions): SetupResult { ... }
```

Reply:
I like it.  I added these types in to make `setup` more explicit.  I put the type declarations _inside_ the `describe` as I thought they should be scoped _just_ to this set of tests on the premise that other eventual tests in this suite might have different `setup`.  LMK if this goes against the spirit of your other suggestion to bump the `Ajax` and `WorkspaceDataProps` up to the top level, or if it makes sense to keep the types local where possible.

Reply:
Fair enough. I don't feel very strongly about the location of types. Since Ajax is "global" for this module, I think it makes sense to move the Ajax type up, but keeping the others here makes sense.